src/clapsolver/clap.py

```diff
@@ -33,3 +33,124 @@ def cost(i, j, costs):
 
     # Total cost is that of the submatrix lsap plus the cost of the constraint.
     return sub_costs[sub_row_ind, sub_col_ind].sum() + costs[i, j]
+
+
+def constrained_lsap_costs(costs):
+    """Solve a constrained linear sum assignment problem for each entry.
+
+    TODO: More thorough testing of this function.
+
+    The output of this function is equivalent to, but significantly more
+    efficient than,
+
+    >>> def constrained_lsap_costs(costs):
+    ...     total_costs = np.empty_like(costs)
+    ...     for i, j in np.ndindex(*costs.shape):
+    ...         total_costs[i, j] = constrained_lsap_cost(i, j, costs)
+    ...     return total_costs
+
+    Parameters
+    ----------
+    costs : 2darray
```

Comment:
`costs` parameter should be renamed to avoid name conflict. Perhaps `cost_matrix`.